### DISCUSSION TITLE
Code works on local, not on server

### DISCUSSION THREAD

**Discussion by cpaschen** [joomengine/Joomla-Component-Builder#818] [Q&A]
I'm developing a rather simple component, but running into a VERY weird problem.

It has a form that allows 3 files to be uploaded.
After the files are uploaded (postSaveHook), the files are processed via a custom function in the main component helper file.

That's it (so far); however, when I run it on my local development environment (J3.10 with JCB on WAMP), everything runs fine.
But when I compile and install on my live server, I can open and complete the form, but after pressing the Save & Close button it just processes for a bit and then returns to the component dashboard. Nothing is saved in the component's database. No errors in the log file.
Absolutely nothing.

I have custom save code added to the model file; however, it doesn't appear that code is even being called

I've completely commented out the functioning from the controller's postSaveHook function area (i.e. links to my function in the helper), as well as all modifications in the model's save function and still nothing is saved, not even a blank record. And no errors either (on screen or in the log file).
I am testing with a SuperUser account so access should not be an issue.

It's like the model/controller for this view aren't being called - but this problem is only happening on the live site, not on the local dev site.
(And I've confirmed that I'm working with the proper component admin views.)

I've never had this happen with a JCB coponent.

I do have other (commercial) extensions installed on this site (same on local dev version as well as live site) and those extensions are all working fine.

Anyone ever had this happen or have any other ideas on tracking down what the issue is?

**Comment by cpaschen** [comment]
Partially tracked this down ...

Issues were:

1. Improper access settings on the component's /media folder, was blocking ability to write files there
2. Server memory and file size uploads (and processing time) - only 3 files but they are pretty huge, so had to increase both length of time to process and available memory size.

I'm still tweaking 2 so that I can get it to fully run, but at least it's getting through part of the processing now.
(Hopefully I won't have to add more memory to the server just to run this component).

**Comment by MastersOfMedia** [comment]
Hi Chris, this makes sense as (as far as I know) files are being stored in memory when being up- or downloaded. So you need to assign enough memory to in order make this work. I’ve had a similar situation where we wanted users to download files (same principal applies) in excess of 2GB and we immediately ran into memory issues. You should be able to track these issues in the server log though, as memory issues normally will be reported in the server logfiles.

Best regards,

Erik van Doorne

**Comment by cpaschen** [comment]
Erik,
Thanks. Did you come up with any methods that minimize the memory usage?

Our process runs once a week and otherwise the site needs very little resources for 'every day' functionality. So we're considering re-writing the code to run locally (do all 'processing offline') and possibly upload it in batches.
But I'd be interested to know if you (or others) have come up with any creative ways to save memory when dealing with large files.

**Reply by Llewellynvdm** [reply]
There are upload and download methods in PHP that use less memory, you will need to do some research on this.

In JCB we mostly use the [file_get_contents](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/helpers/componentbuilder.php#L4275) and in most cases we get away with moving large files. Yet there is a streaming option in the framework, here is the [documentation](https://api.joomla.org/cms-3/classes/Joomla.Filesystem.Stream.html#method_getStream), and here is the [code](https://github.com/joomla-framework/filesystem).

You will see that the steaming class has a [fallback](https://github.com/joomla-framework/filesystem/blob/2.0-dev/src/Stream.php#L316) to fopen. Reality is that you can learn much from just reading over the Joomla filesystem classes.
